### PR TITLE
Issue-1113: Create hello-strongbox-nuget-chocolatey

### PR DESCRIPTION
**Pull request by TheCakeIsNaOH**
From [Strongbox #1113](https://github.com/strongbox/strongbox/issues/1113)

Requires to strongbox/strongbox#1709

### CODE REVIEW

**Comment by carlspring** [comment]
Hi @TheCakeIsNaOH,

The text looks great, thank you! Would you mind also adding all the files that were involved as well?

**Comment by TheCakeIsNaOH** [comment]
All other files are created either by the `choco new` or `choco pack` commands. If you would like, I can change it so that the `choco new` command is not used, and instead include the output files from `choco new`.

**Comment by steve-todorov** [comment]
@carlspring / @TheCakeIsNaOH can we now merge this? It looks like we've missed to do so? :)

**Comment by carlspring** [comment]
I don't think we can, due the bug with artifacts having hyphens in their paths.
@TheCakeIsNaOH : Did you file a bug in the Strongbox project for this? If you file a bug with the description, perhaps @cbrianhill and/or @sbespalov could look into it and work with you to resolve it, so that we could merge this pull request? What do you guys think?

**Comment by TheCakeIsNaOH** [comment]
@steve-todorov as carlspring mentioned it should not be merged till the install bug is resolved. Additionally, it probably would be best to wait until the chocolatey master has the fixed mono build (chocolatey/choco#1827), they just take a long time to review PRs there.

@carlspring Will do on filing a separate report.

**Comment by steve-todorov** [comment]
Thanks for the update @TheCakeIsNaOH! I've dropped a message in chocolatey/choco#1827. Hopefully your fixes get merged soon :)

File: hello-strongbox-nuget-chocolatey/README.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+It will create a folder named `hello-chocolatey` with some files including a `.nuspec` and a folder called tools.
+
+2. Delete `_TODO.txt` and `ReadMe.md` in the hello-chocolatey directory and also  
+
+3. Delete `chocolateybeforemodify.ps1` and `chocolateyuninstall.ps1` in the tools sub-directory
+
+3. Edit `chocolateyinstall.ps1` in the tools sub-directory with a text editor
+
+4. Remove all text in the file and replace with `write-host Package would install here`
```
Comment (steve-todorov) [inline hello-strongbox-nuget-chocolatey/README.md]: Could you please add an exact example of how the `tools/chocolateyinstall.ps1` file should look and if there is anything special about it? (i.e. some tools really don't like files to end with a new line, but others specifically require that)

File: hello-strongbox-nuget-chocolatey/README.md [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+
+
+## How to make Chocolatey NuGet package
+
+Execute the following command in the same directory as hello-chocolatey.nuspec:
+```sh
+choco pack
+```
```
Comment (steve-todorov) [inline hello-strongbox-nuget-chocolatey/README.md]: I am not able to pass this step:
```
bash-5.0$ cat tools/chocolateyinstall.ps1 
write-hostmcediackage would install here
(file ends with a new line)
bash-5.0$ choco pack
Chocolatey v0.10.16.0
Chocolatey is not an official build (bypassed with --allow-unofficial).
 If you are seeing this message and it is not expected, your system may
 now be in a bad state. Only official builds are to be trusted.

Attempting to build package from 'hello-chocolatey.nuspec'.
Cannot create a package that has no dependencies nor content.
(exits)
```

I'm using our new docker image from strongbox/ci-build-images#8 which contains a build of `chocolatey/choco#9bb4246`. A snapshot of the linux image is available in [Docker Hub](https://hub.docker.com/layers/strongboxci/alpine/jdk8-mvn3.6-mono5-nuget3.4-choco0.10-PR-8/images/sha256-388010cdecd4704347b33bed6ff94a0db7b37328d9b0c7bb173eaec640147c1f?context=repo) and you can use it: 
```
docker run -it --rm strongboxci/alpine:jdk11-mvn3.6-mono5-nuget3.4-choco0.10-PR-8
```

File: hello-strongbox-nuget-chocolatey/README.md (hunk 1 above)
Comment (TheCakeIsNaOH) [reply hello-strongbox-nuget-chocolatey/README.md]: I have added an example.

File: hello-strongbox-nuget-chocolatey/README.md (hunk 2 above)
Comment (TheCakeIsNaOH) [reply hello-strongbox-nuget-chocolatey/README.md]: I added a note about it, see https://github.com/strongbox/strongbox-docs/pull/78

**Comment by TheCakeIsNaOH** [comment]
Moved contents of readme to- https://github.com/strongbox/strongbox-docs/pull/78
as per https://github.com/strongbox/strongbox/issues/1580

**Review by steve-todorov — APPROVED** [review]
LGTM! Thank you very much @TheCakeIsNaOH for your help with this! Proceeding with merging :)

**Pull request merged**